src/Illuminate/Collections/Collection.php

```diff
@@ -521,6 +521,19 @@ public function implode($value, $glue = null)
         return implode($value ?? '', $this->items);
     }
 
+    /**
+     * Insert a value between each item in the collection.
+     * 
+     * For example: ['a', 'b', 'c'] with 'x' becomes ['a', 'x', 'b', 'x', 'c'].
+     *
+     * @param  string  $value
+     * @return static
+     */
+    public function insertBetweenEach($value)
+    {
+        return new static($this->flatMap(fn ($item) => [$item, $value])->slice(0, -1));
```

Comment:
this seems pretty niche. why not just wrote `->flatMap(fn ($item) => [$item, $value])->slice(0, -1)` in your app?

Reply:
@GrahamCampbell because it's not very clear what it's doing

Reply:
then register a macro in your app?

Reply:
Yes that's possible but I think others can benefit from this as well